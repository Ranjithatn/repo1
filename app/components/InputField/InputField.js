import React from "react";
import Input from "../Input/Input";
import Label from "../Label/Label";

const InputField = ({
  labelId,
  labelName,
  labelClassName,
  labelText,
  inputId,
  inputClassName,
  value,
  onChange,
  inputName,
  inputType,
  placeholder,
  helpMsg,
  helpClassName,
  disabled = false,
  onKeyPress,
}) => {
  return (
    <div className="field">
      {labelText && (
        <Label
          className={labelClassName ? labelClassName : ""}
          text={labelText}
          name={labelName ? labelName : ""}
          htmlFor={inputId ? inputId : ""}
        />
      )}
      <div className="control">
        <Input
          className={inputClassName ? inputClassName : ""}
          placeholder={placeholder ? placeholder : ""}
          type={inputType ? inputType : "text"}
          inputClassName={inputClassName}
          value={value}
          id={inputId ? inputId : ""}
          name={inputName ? inputName : ""}
          onChange={onChange}
          disabled={disabled}
          onKeyPress={ onKeyPress ? onKeyPress : () => {} }
        />
      </div>
      {helpMsg && (
        <p className={"help" + (helpClassName ? " " + helpClassName : "")}>
          {helpMsg}
        </p>
      )}
    </div>
  );
};

export default InputField;
