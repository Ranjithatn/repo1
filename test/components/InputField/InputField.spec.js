import React from "react";
import toJson from "enzyme-to-json";
import InputField from "../../../app/components/InputField/InputField";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });


describe('<InputField />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <InputField />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
