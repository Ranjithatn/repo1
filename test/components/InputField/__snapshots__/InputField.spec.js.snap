// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> should render without crashing 1`] = `
<InputField>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <Input
        className=""
        disabled={false}
        id=""
        name=""
        onKeyPress={[Function]}
        placeholder=""
        type="text"
      >
        <div
          className="control"
          style={Object {}}
        >
          <input
            className=""
            disabled={false}
            id=""
            name=""
            onKeyPress={[Function]}
            placeholder=""
            style={Object {}}
            type="text"
          />
        </div>
      </Input>
    </div>
  </div>
</InputField>
`;
